net/http/authn/context.go

```diff
@@ -0,0 +1,36 @@
+package authn
+
+import "context"
+
+type key int
+
+const (
+	tokenKey key = iota
+	localhostKey
+)
+
+// newContextWithToken sets the token in a new context and returns the context.
+func newContextWithToken(ctx context.Context, token string) context.Context {
+	return context.WithValue(ctx, tokenKey, token)
+}
+
+// TokenFromContext returns the token stored in the context, if there is one.
+func TokenFromContext(ctx context.Context) (string, bool) {
```

Comment:
🚲 🏠:

- Should these accessors should be named `Token` and `Localhost`?
- Should this function return just a `string` (the empty string if there was no valid token)?

Then the usage might look like

```go
func authzToken(ctx context.Context) bool {
	// TODO(tessr): compare against Policies
	return authn.Token(ctx) != ""
}
```